contracts/common/DomainTokenReceiver.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.8.27; // ignore-swc-103
+import { ERC20Extended } from "./ERC20Extended.sol";
+import { DSAuth } from "./../../lib/dappsys/auth.sol";
+
+contract DomainTokenReceiver is DSAuth {
+  address resolver; // Storage slot 2 (from DSAuth there is authority and owner at storage slots 0 and 1 respectively)
+
+  address colony;
+
+  modifier onlyColony() {
+    require(msg.sender == colony, "domain-token-receiver-unauthorized");
+    _;
+  }
+
+  function getColonyAddress() public view returns (address) {
+    return colony;
+  }
+
+  function setColonyAddress(address _colony) public auth {
+    require(colony == address(0), "domain-token-receiver-colony-already-set");
+    colony = _colony;
+  }
+
+  function transferNativeToColony() public onlyColony {
```

Comment:
Can we call this `transferChainNativeToColony`? I keep getting mixed up between the two senses of "native token"